File: leetcode/0589. N-ary Tree Preorder Traversal/Solution.swift
// https://leetcode.com/problems/n-ary-tree-preorder-traversal/

public class TreeNode {
    public var val: Int
    public var children: [TreeNode?]
    public init(_ val: Int) {
        self.val = val
        self.children = []
    }
}

class Solution {
    func preorder(_ root: TreeNode?) -> [Int] {
        guard let root = root else { return [] }

        var r = [root.val]
        for child in root.children {
            r += preorder(child)
        }

        return r
    }
}

File: leetcode/0589. N-ary Tree Preorder Traversal/Solution.py
# https://leetcode.com/problems/n-ary-tree-preorder-traversal/

"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def preorder(self, root: 'Node') -> List[int]:
        if root is None:
            return []
        
        r = [root.val]
        for child in root.children:
            r += self.preorder(child)
        
        return r